### PR TITLE
Improve factories

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**

### CODE REVIEW

File: src/main/java/net/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java [hunk 1]
```diff
@@ -135,15 +125,15 @@ public void updateSkin(Player player) {
     private void checkOptFile() {
         File fileDisableDismountPlayer = new File("plugins" + File.separator + "SkinsRestorer" + File.separator + "disablesdismountplayer");
         File fileEnableDismountEntities = new File("plugins" + File.separator + "SkinsRestorer" + File.separator + "enablesdismountentities");
-        File fileEnableRemountEntiteis = new File("plugins" + File.separator + "SkinsRestorer" + File.separator + "enablesremountentities");
+        File fileEnableRemountEntities = new File("plugins" + File.separator + "SkinsRestorer" + File.separator + "enablesremountentities");
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java:128]: https://discord.com/channels/186794372468178944/815969158470697021/818415205099765802

**Review by xknat — CHANGES_REQUESTED** [review]
As discussed in discord:

>also, i checked config.java
we did switch REMOUNT_PLAYER_ON_UPDATE 
so enablesremountentities should be disablesremountplayer & disablesremoun.exist should turn it to false

**Review by xknat — APPROVED** [review]

**Review by xknat — CHANGES_REQUESTED** [review]
These changes will bring the current mc 1.16 custom world name issue to lower versions.
Something we would like to have fixed first before merging.

File: src/main/java/net/skinsrestorer/bukkit/skinfactory/OldSkinRefresher.java [hunk 2]
```diff
@@ -117,47 +118,36 @@ public void accept(Player player) {
...
+                                int.class, peaceful.getClass(), worldType.getClass(), enumGamemode.getClass()
                         },
-                        dimension, difficulty, worldtype, ReflectionUtil.invokeMethod(enumGamemode.getClass(), null, "getById", new Class<?>[]{int.class}, gamemodeId));
+                        dimension, difficulty, worldType, ReflectionUtil.invokeMethod(enumGamemode.getClass(), null, "getById", new Class<?>[]{int.class}, gamemodeId));
             } catch (Exception ignored) {
-                if (environment.equals(World.Environment.NETHER))
+                if (environment == World.Environment.NETHER)
                     dimension = -1;
-                else if (environment.equals(World.Environment.THE_END))
+                else if (environment == World.Environment.THE_END)
```
Comment (AlexProgrammerDE) [inline src/main/java/net/skinsrestorer/bukkit/skinfactory/OldSkinRefresher.java:150]: @xknat the bug is somewhere caused here.

**Comment by AlexProgrammerDE** [comment]
Decided to be made for v14.1 instead.

**Pull request merged**